sdks/python/apache_beam/metrics/base.py

```diff
@@ -0,0 +1,121 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+The classes in this file are base classes for metrics. They are not intended
+to be subclassed or created directly by users. To work with and access metrics,
+ users should use the classes and methods exposed in metric.py.
+
+Available classes:
+- Metric - Base class of a metrics object. Provides very basic methods.
+- Counter - Base class of a counter metric object. It provides methods to
+    increment/decrement a count variable accross a pipeline execution.
+- Distribution - Base class of a distribution metric object. It provides
+    methods to keep track of statistics about the distribution of a variable.
+- MetricName - Base name of a metric: (namespace=str, name=str)
+- MetricKey - Base internal key of a metric
+    (step_name=str, metric=<MetricName>)
+- MetricResult - Current status of a metric's updates/commits
+    (key=<MetricKey>, committed=data, attempted=data)
```

Comment:
This should is not really a base class for metrics; rather, it is used for execution and corresponds more to the classes defined in "internal.py".